### PR TITLE
Correct test flow for TC delete_volume_on_deleted_vm

### PR DESCRIPTION
**Pull request by albinsun**
## Changes
1.Fix wrong test flow for TC `test_delete_volume_on_deleted_vm`.

## Verification
`harvester-runtests/8` on ECM lab
image

### CODE REVIEW

File: harvester_e2e_tests/integrations/test_1_volumes.py [hunk 1]
```diff
@@ -178,32 +178,26 @@ def test_delete_volume_on_existing_vm(self, api_client, ubuntu_image, ubuntu_vm,
...
-        code, data = api_client.volumes.delete(vol_name)
-        assert 422 == code, f"Should fail to delete volume\n{code}, {data}"
-
-        self.delete_vm(api_client, ubuntu_vm, polling_for)
+        api_client.vms.delete(vm_name)
 
-        code, data = api_client.volumes.delete(vol_name)
-        assert code in (200, 204), f"Fail to delete volume\n{code}, {data}"
+        polling_for("Delete volume",
+                    lambda c, d: c and c != 422,
```
Comment (lanfon72) [inline harvester_e2e_tests/integrations/test_1_volumes.py:194]: what's the different to check `200` and `422`?

Comment (albinsun) [reply harvester_e2e_tests/integrations/test_1_volumes.py:194]: We can catch the case that fail to delete due to still attaching on VM.

Comment (albinsun) [reply harvester_e2e_tests/integrations/test_1_volumes.py:194]: Per discussion, `422` seems the coupled part between VM and volume. But yeh, I think we can just assert `200` for more explicit logic.

Comment (lanfon72) [reply harvester_e2e_tests/integrations/test_1_volumes.py:194]: I feel that `422` would need a comment to explain it is going to check that the volume is detached, `200` would simply self-explained that the volume is deleted successfully.

**Review by lanfon72 — APPROVED** [review]

**Pull request merged**